### PR TITLE
Add Lua support

### PR DESCRIPTION
**Pull request by krenzland**
This PR adds initial support for Lua in easi.
Lua is a popular scripting language and often used e.g. for scripting in video games.
It is versatile, very very simple and runs on nearly every platform.
I've also added an example to this.

**Note:** This version is not threadsafe but easi is not usually used from a threaded background. Is this a problem? If yes, we can circumvent this by using one Lua state per thread.

### CODE REVIEW

**Review by sebwolf-de — APPROVED** [review]
Hi, LGTM, just one question about the example.
About the thread safety: I think we can merge this now, if anyone encounters a problem, we can still add the thread saftiness...

File: examples/4_lua.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    components: !ConstantMap
+      map:
+        rho:     2500.0
+        mu:         0.0
+        lambda: 1.96e10
+  - !LuaMap
+      returns: ["rho", "mu", "lambda"]
+      function: |
+         function f (x)
+          io.write(x[1], " ", x[2], " ", x[3], "\n")
```
[the remark this answers is not in the document]
Comment (krenzland) [reply examples/4_lua.yaml:14]: It's not needed - it's just a demonstration that this is possible - you can debug it more easily. I can also delete it if you think it's clearer.

[the remark this answers is not in the document]
Comment (sebwolf-de) [reply examples/4_lua.yaml:14]: Maybe add a comment?

[the remark this answers is not in the document]
Comment (krenzland) [reply examples/4_lua.yaml]: Good idea, added.

**Comment by sebwolf-de** [comment]
Good that you updated the version number, I did not notice that.

**Pull request merged**